### DISCUSSION TITLE
Some Questions

### DISCUSSION THREAD

**Discussion by mattbha** [wrabit/django-cotton#173] [General]
Couldn't find documentation for these, apologies and please link me if it exists. Perhaps I am using things incorrectly.
_for reference i am on django-cotton (1.0.4)_

**1) nested `attrs` don't seem to be passed all the way down**

`my_view.html`
--------------
`<c-arrow class="absolute right-3 top-1/2 transform -translate-y-1/2 h-5 w-5 text-gray-500" />`

`arrow.html`
--------------
```
<!-- forward arrow -->
<c-svg {{ attrs }}>
    <path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M9 5l7 7-7 7" />
</c-svg>
```

`svg.html`
--------------
```
<svg
    xmlns="http://www.w3.org/2000/svg"
    fill="none"
    stroke="currentColor"
    preserveAspectRatio="xMidYMid meet"
    {{ attrs }}>
    {{ slot }}
</svg>
```

it doesn't appear the `class=...` will make it to svg.html's `{{ attrs }}`. I would expect it would, because arrow.html has ```c-svg {{ attrs }}``` that attrs (class=...) be passed to svg.html as attrs

-------------------------------------------------------------------------------------------------
2) is `class` not sent in attrs?

seems like i need to do
```<input type="text" {{ attrs }} class="{{ class }}">```

rather than just 

```<input type="text" {{ attrs }}>```

assuming the calling function adds a `class="xyz"` attr

**Comment by mattbha** [comment]
I've ended up working around both the above, and do not think they should be used to drive any changes necessarily.

**Comment by wrabit** [comment]
1. Currently, we are limited to what we can throw into the <c- tags as these compile to native django (so your example above compiles to `{% c svg {{ attrs }} %}`. But I think it would be a worthy use case to support. 

2. Class should not be treated any differently, only if you specified it as a cvar.

(About the attribute casing issue, it should retain casing, it only modifies '-' to '_' to make the var accessible)

**Comment by mattbha** [comment]
@wrabit  is `only` supported (like a django template include)
if not, could be nice to have (especially for testing/debugging)

**Reply by mattbha** [reply]
i went ahead and made a PR for this https://github.com/wrabit/django-cotton/pull/174

**Comment by wrabit** [comment]
Proxying and passing {{ attrs }} dict now supported in [2.1.0](https://github.com/wrabit/django-cotton/releases/tag/2.1.0)
